File: parasmittal/Day 18 & 19/Longest Repeating Character Replacement.cpp
class Solution {
public:
    int replace(string s,int k,char c){
        int count=0;
        int len=0;
        int start=0;
        int end=0;
        while(end<s.length()){
            if(s[end]!=c){
                count++;
                while(count>k){
                    if(s[start]!=c)
                        count--;
                    start++;
                }
                
            }
            len=max(len,end-start+1);
            end++;
        }
        return len;
    }
    int characterReplacement(string s, int k) {
        int len=0;
        for(int i=0;i<26;i++){
            len=max(len,replace(s,k,'A'+i));
        }
        return len;
    }
};

File: parnamondal/Week -4/Day -2 [Pro Mode]/Longest Repeating Character Replacement.py
# Time Complexity - O(N)
# Space Complexity - O(1)

class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        co=[0]*26
        ans,curr,left,size,count=0,0,0,0,0
        for right in range(len(s)):
            co[ord(s[right])-65]+=1 
            size=right-left+1
            count=max(count,co[ord(s[right])-65])
            curr=size-count
            if curr>k:
                co[ord(s[left])-65]-=1
                left+=1
            else:
                ans=max(ans,size)
        return ans
        